### PR TITLE
Fix Robin errors

### PR DESCRIPTION
**Pull request by guyer**
Address some issues observed by @cashTangoTangoCash in [RobinBoundaryConditionsFiPyUsageConvectionBC.pdf](https://github.com/cashTangoTangoCash/Convection2DFiPyExample01/blob/master/RobinBoundaryConditionsFiPyUsageConvectionBC.pdf)

### CODE REVIEW

File: documentation/USAGE.rst [hunk 1]
```diff
@@ -575,7 +575,7 @@ can often be substituted for the flux in an equation
 >>> diffusionCoeff = FaceVariable(mesh=mesh, value=b)
 >>> diffusionCoeff.setValue(0., where=mask)
 >>> eqn = (TransientTerm() == PowerLawConvectionTerm(coeff=convectionCoeff)
->>>        + DiffusionTerm(coeff=diffusionCoeff) + (g * mask).divergence)
+>>>        + DiffusionTerm(coeff=diffusionCoeff) + (g * mask * mesh.faceNormals).divergence)
```
Comment (tkphd) [inline documentation/USAGE.rst]: In the built PDF, this wraps confusingly. Recommend replacing with
```
>>> eqn = (TransientTerm() == PowerLawConvectionTerm(coeff=convectionCoeff) +
>>>        + DiffusionTerm(coeff=diffusionCoeff)
>>>        + (g * mask * mesh.faceNormals).divergence)
```

File: documentation/USAGE.rst [hunk 2]
```diff
@@ -641,7 +641,7 @@ can be constrained to have a Robin condition at a face identifed by
 >>> Gamma.setValue(0., where=mask)
 >>> dPf = FaceVariable(mesh=mesh, value=mesh._faceToCellDistanceRatio * mesh.cellDistanceVectors)
 >>> Af = FaceVariable(mesh=mesh, value=mesh._faceAreas)
->>> RobinCoeff = (mask * Gamma0 * Af / (dPf.dot(a) + b)).divergence
+>>> RobinCoeff = (mask * Gamma0 * Af * mesh.faceNormals / (dPf.dot(a) + b)).divergence
 >>> eqn = (TransientTerm() == DiffusionTerm(coeff=Gamma)
 ...        + RobinCoeff * g - ImplicitSourceTerm(coeff=RobinCoeff * mesh.faceNormals.dot(a)))
```
Comment (tkphd) [inline documentation/USAGE.rst]: Unpleasant line-wrapping in built PDF. Recommend replacing with
```
>>> eqn = (TransientTerm() == DiffusionTerm(coeff=Gamma) + RobinCoeff * g
...        - ImplicitSourceTerm(coeff=RobinCoeff * mesh.faceNormals.dot(a)))
```

**Comment by guyer** [comment]
@tkphd Thanks for the suggesting line wrappings. I've made some more pervasive changes to hopefully make this a bit clearer. Please take another look.

File: documentation/USAGE.rst [hunk 3]
```diff
@@ -558,108 +558,118 @@ can often be substituted for the flux in an equation
```
Comment (tkphd) [inline documentation/USAGE.rst]: The condition `on $f=f_0$` is first used at the top of this section, but neither term is ever defined.

File: documentation/USAGE.rst [hunk 4]
```diff
@@ -558,108 +558,118 @@ can often be substituted for the flux in an equation
...
-The Robin condition can be used to substitute for the expression 
-:math:`\left(\hat{n}\cdot\nabla\phi\right)_{f_0}`
-but we note that :term:`FiPy` calculates variable values at cell centers 
-and gradients at intervening faces. We obtain a first-order approximation 
-for :math:`\left(\hat{n}\cdot\nabla\phi\right)_{f_0}` in terms of 
-neighboring cell values by substituting 
+where :math:`\vec{d}_{fP}` is the distance vector from the face center to
+the adjoining cell center.  The approximation
+:math:`\left(\vec{d}_{fP}\cdot\nabla\phi\right)_{f_0} \approx
+\left(\hat{n}\cdot\nabla\phi\right)_{f_0}\left(\vec{d}_{fP}\cdot\hat{n}\right)_{f_0}`
```
Comment (tkphd) [inline documentation/USAGE.rst]: Good caveat.

File: documentation/USAGE.rst [hunk 5]
```diff
@@ -558,108 +558,118 @@ can often be substituted for the flux in an equation
...
 
-   \begin{aligned}
-        \hat{n}\cdot\left(\vec{a} \phi + b \nabla\phi\right)_{f_0} &= g \\
-        \hat{n}\cdot\left(\vec{a} \phi_P
-        - \vec{a} \left(\hat{n}\cdot\nabla\phi\right)_{f_0}\left(\vec{d}_{fP}\cdot\hat{n}\right)_{f_0}
-        + b \nabla\phi\right)_{f_0} &\approx g \\
-        \left(\hat{n}\cdot\nabla\phi\right)_{f_0} 
-        &\approx \frac{g - \hat{n}\cdot\vec{a} \phi_P}{-\left(\vec{d}_{fP}\cdot\vec{a}\right)_{f_0} + b}
-   \end{aligned}
+to obtain the value on the bounary face in terms of the neighboring cell.
```
Comment (tkphd) [inline documentation/USAGE.rst]: bounary --> boundary

File: documentation/USAGE.rst [hunk 6]
```diff
@@ -558,108 +558,118 @@ can often be substituted for the flux in an equation
...
 
-can be constrained to have a Robin condition at a face identifed by 
+can be constrained to have a Robin condition at a face identifed by
 ``mask`` by making the following modifications
 
 >>> Gamma = FaceVariable(mesh=mesh, value=Gamma0)
 >>> Gamma.setValue(0., where=mask)
->>> dPf = FaceVariable(mesh=mesh, value=mesh._faceToCellDistanceRatio * mesh.cellDistanceVectors)
+>>> dPf = FaceVariable(mesh=mesh, 
+...                    value=mesh._faceToCellDistanceRatio * mesh.cellDistanceVectors)
```
Comment (tkphd) [inline documentation/USAGE.rst]: Much prettier.

File: documentation/USAGE.rst [hunk 7]
```diff
@@ -558,108 +558,118 @@ can often be substituted for the flux in an equation
...
 >>> Af = FaceVariable(mesh=mesh, value=mesh._faceAreas)
->>> RobinCoeff = (mask * Gamma0 * Af / (dPf.dot(a) + b)).divergence
->>> eqn = (TransientTerm() == DiffusionTerm(coeff=Gamma)
-...        + RobinCoeff * g - ImplicitSourceTerm(coeff=RobinCoeff * mesh.faceNormals.dot(a)))
+>>> RobinCoeff = (mask * Gamma0 * Af * mesh.faceNormals / (dPf.dot(a) + b)).divergence
+>>> eqn = (TransientTerm() == DiffusionTerm(coeff=Gamma) + RobinCoeff * g
+...        - ImplicitSourceTerm(coeff=RobinCoeff * mesh.faceNormals.dot(a)))
```
[the remark this answers is not in the document]
Comment (wd15) [reply documentation/USAGE.rst]: or `type(a)` or maybe set a value for `a`

[the remark this answers is not in the document]
Comment (wd15) [reply documentation/USAGE.rst]: It's useful to know what `a` and `b` are as FiPy or Python objects.

File: documentation/USAGE.rst (hunk 3 above)
Comment (guyer) [reply documentation/USAGE.rst]: I completely changed  how the boundary faces are identified, both to make clearer and  more rigorous (I think)

**Comment by guyer** [comment]
@tkphd & @wd15:  I  made more than minor changes to address your comments. Please check.

**Pull request merged**